homeassistant/components/zabbix/__init__.py

```diff
@@ -46,13 +90,208 @@ def setup(hass, config):
     username = conf.get(CONF_USERNAME)
     password = conf.get(CONF_PASSWORD)
 
-    zapi = ZabbixAPI(url)
+    publish_states_host = conf.get(CONF_PUBLISH_STATES_HOST)
+
+    include = conf.get(CONF_INCLUDE, {})
+    exclude = conf.get(CONF_EXCLUDE, {})
+    whitelist_e = set(include.get(CONF_ENTITIES, []))
+    whitelist_d = set(include.get(CONF_DOMAINS, []))
+    blacklist_e = set(exclude.get(CONF_ENTITIES, []))
+    blacklist_d = set(exclude.get(CONF_DOMAINS, []))
+
+    max_tries = conf.get(CONF_RETRY_COUNT)
+
     try:
-        zapi.login(username, password)
+        zapi = ZabbixAPI(url=url, user=username, password=password)
         _LOGGER.info("Connected to Zabbix API Version %s", zapi.api_version())
     except ZabbixAPIException as login_exception:
         _LOGGER.error("Unable to login to the Zabbix API: %s", login_exception)
         return False
+    except HTTPError as http_error:
+        _LOGGER.error("HTTPError when connecting to Zabbix API: %s", http_error)
+        zapi = None
+        # Continue even though Zabbix API couldn't be initialized. Zabbix sender later may still work.
 
     hass.data[DOMAIN] = zapi
+
+    def event_to_metrics(event, float_keys, string_keys):
+        """Add an event to the outgoing Zabbix list."""
+        state = event.data.get("new_state")
+        if (
+            state is None
+            or state.state in (STATE_UNKNOWN, "", STATE_UNAVAILABLE)
+            or state.entity_id in blacklist_e
+            or state.domain in blacklist_d
+        ):
+            return
+
+        entity_id = state.entity_id
+        floats = {}
+        strings = {}
+        try:
+            if (
```

Comment:
Please move this check out of the try... except block.

Reply:
Ok